rclcpp/src/rclcpp/executors/executor_notify_waitable.cpp

```diff
@@ -60,6 +60,10 @@ ExecutorNotifyWaitable::add_to_wait_set(rcl_wait_set_t & wait_set)
     rcl_guard_condition_t * cond = &guard_condition->get_rcl_guard_condition();
     rcl_ret_t ret = rcl_wait_set_add_guard_condition(&wait_set, cond, NULL);
 
+    if(was_triggered_) {
+      guard_condition->trigger();
+    }
+
```

Comment:
Can you explain why this particular waitable needs to do this? I explicitly didn't update this one because it did not appear to need it. It was an example of a waitable that was only ever triggered in order to wake up the wait set, and so it didn't matter if it was actually executed or not, and in fact it would lead to spurious wake ups of the executor if this was triggered but not handled immediately.

I'm not actually convinced this should be done here.

Reply:
Changed the whole approach, as discussed with @mjcarroll I added special case handling for  notify_waitable_ to be checked and executed directly in wait_for_work